src/components/Dashboard/functions.js

```diff
@@ -0,0 +1,119 @@
+// and https://stackoverflow.com/questions/19014250/rerender-view-on-browser-resize-with-react/19014495#19014495
+export const useDashboardSettings = ({ marginRight = 16, dashboardEl }) => {
+    const [dimensions, setDimensions] = useState({
+        height: 0,
+        width: 0,
+        paddingLeft: 0,
+        paddingRight: 0,
+    });
+    const [cardWidth, setCardWidth] = useState(0);
+    const [cardHeight, setCardHeight] = useState(0);
+    const [numColumns, setNumColumns] = useState(0);
+
+    useIsomorphicLayoutEffect(() => {
+        function updater() {
+            if (dashboardEl.current) {
+                setDimensions({
+                    width: dashboardEl.current.offsetWidth,
+                    height: dashboardEl.current.offsetHeight,
+                    paddingLeft: parseFloat(
+                        window
+                            .getComputedStyle(dashboardEl.current)
+                            .getPropertyValue("padding-left")
+                    ),
+                    paddingRight: parseFloat(
+                        window
+                            .getComputedStyle(dashboardEl.current)
+                            .getPropertyValue("padding-right")
+                    ),
+                });
+            }
+        }
+        window.addEventListener("resize", updater);
+        updater();
+        return () => window.removeEventListener("resize", updater);
+    }, [dashboardEl, setDimensions]);
+
+    // This is used because media queries misbehave on the server and this lets
+    // us set values before rendering occurs.
+    useIsomorphicLayoutEffect(() => {
+        const { width, paddingLeft, paddingRight } = dimensions;
+
+        let newColumns;
+
+        if (width >= constants.XL_PIXELS) {
+            newColumns = constants.XL_NUM_COLUMNS;
+        } else if (width >= constants.LG_PIXELS) {
+            newColumns = constants.LG_NUM_COLUMNS;
+        } else if (width >= constants.MD_PIXELS) {
+            newColumns = constants.MD_NUM_COLUMNS;
+        } else if (width >= constants.SM_PIXELS) {
+            newColumns = constants.SM_NUM_COLUMNS;
+        } else if (width >= constants.XS_PIXELS) {
+            newColumns = constants.XS_NUM_COLUMNS;
+        } else {
+            // probably won't get here. probably.
+            newColumns = constants.LG_NUM_COLUMNS;
+        }
+
+        const actualWidth = width - paddingLeft - paddingRight;
+
+        // Yeah, the 6 is magical. Haven't figured out why it's needed yet.
+        const cardWidth =
+            newColumns > 1
+                ? Math.floor(actualWidth / newColumns - marginRight) - 6
+                : actualWidth;
+
+        // Try to get a 4:3 ration for width to height
+        const cardHeight = Math.floor(cardWidth - cardWidth / 3);
+
+        setCardWidth(cardWidth);
+        setCardHeight(cardHeight);
+        setNumColumns(newColumns);
+    }, [dimensions, marginRight, setCardWidth, numColumns, setNumColumns]);
```

Comment:
This is a lot of sizing and calculations 😄 And I'm sure the answer is no, but did `Grid` not work?